dvc/repo/ls.py

```diff
@@ -0,0 +1,99 @@
+import os
+
+
+@staticmethod
+def ls(url, targets=[], rev=None, recursive=None):
+    if os.path.exists(url):
+        return _ls_local_repo(url, targets, recursive)
+
+    return _ls_git_repo(url, targets, rev, recursive)
+
+
+def _ls_files_local_repo(path, targets=[], recursive=None):
+    from dvc.repo import Repo
+
+    repo = Repo(path)
+    files = []
+    if not len(targets):
+        targets.append(".")
+
+    def prettify(prefix, filename, is_root):
+        if not is_root:
+            filename = os.path.join(prefix, filename)
+
+        if os.path.isdir(filename):
```

Comment:
Using `os.path` works here, but using it is not correct, strictly speaking :slightly_smiling_face:  as you are using `repo.tree` which might not exist in the real world. `walk` provides you with files and dirs separately, so there is really no need to call any `isdir` here.

Reply:
I  wanted to split out `prettify` from loop logic like the function accepts  the filename and prettify it
I can create 2 functions as
```
def prettify_file(...)

def prettify_dir(...)